charm4py/charm.py

```diff
@@ -166,6 +174,65 @@ def __init_profiling__(self):
         # chares created on this PE
         self.activeChares = set()
 
+    
+    def print_dbg(self, *args, **kwargs):
+        print("PE", self.myPe(), ":", *args, **kwargs)
+    
+    @entry_method.coro
+    def get_future_value(self, fut):
+        #self.print_dbg("Getting data for object", fut.id)
+        obj = fut.lookup_object()
+        if obj == None:
+            local_f = LocalFuture()
+            self.future_get_buffer[fut.store_id] = (local_f, fut)
+            fut.request_object()
+            local_f.get()
+            return fut.lookup_object()
+        else:
+            return obj
+        
+    @entry_method.coro
+    def getany_future_value(self, futs, num_returns):
+        ready_count = 0
+        ready_list = []
+        not_local = []
+        for f in futs:
+            if f.is_local():
+                ready_count += 1
+                ready_list.append(f)
+            else:
+                f.request_object()
+                not_local.append(f)
+        if ready_count >= num_returns:
+            return ready_list[:ready_count]
+        else:
+            local_f = LocalMultiFuture(num_returns - ready_count)
+            for f in not_local:
+                self.future_get_buffer[f.store_id] = (local_f, f)
+            result = local_f.get()
+            for f in not_local:
+                self.future_get_buffer.pop(f.store_id, None)
+            return ready_list + result
+        
+    def check_futures_buffer(self, obj_id):
+        if obj_id in self.future_get_buffer:
+            local_f, fut = self.future_get_buffer.pop(obj_id)
+            local_f.send(fut)
+
+    def check_send_buffer(self, obj_id):
+        completed = self.send_buffer.check(obj_id)
+
+    def check_receive_buffer(self, obj_id):
```

Comment:
Avoid this call when ray.get on a future is called.
Replace future with object in the check function